### ISSUE TITLE
`prefer-expect-assertions` does not evaluate content of `expect.assertions()`

### ISSUE THREAD

**Issue by mad-it** [jest-community/eslint-plugin-jest#830]
Content of the parameter in `expect.assertions()` is not evaluated.

Code sample:
```ts
  it.each<[boolean]>([[true], [false]])(
    'maybe throws error',
    (shouldThrow: boolean): void => {
      expect.assertions(shouldThrow ? 1 : 0); // <== Output is a number, but it gives a warning anyways.

      try {
        doSomething(shouldThrow);
      } catch (error: unknown) {
        expect(error).toMatchSnapshot();
      }
    },
  );
```

**Comment by G-Rath** [comment]
That's because we're a linter, not a static analyser so we can't evaluate the type of expressions aside from basic literal type checks - while your example is relatively simple, it would be very easy to make more advance cases that we couldn't handle.

Additionally, your doing a pattern we recommend not doing: using conditionals in tests.
